File: designpatterns/src/com/renho/create/builder/Product.java
package com.renho.create.builder;

import java.util.ArrayList;
import java.util.List;

public class Product {

	private List<String> parts = new ArrayList<String>();
	
	public void add(String part) {
		this.parts.add(part);
	}
	
	public void show() {
		for(String s:parts) {
			System.out.println(s);
		}
	}
}

File: ideaworkspace/designpatterns/src/main/scala/com/renho/create/builder/Product.scala
package com.renho.create.builder

import scala.collection.mutable.ListBuffer

/**
  * Created by renho on 2016/7/8.
  */
class Product {
	private val parts: ListBuffer[String] = new ListBuffer
	def add(str: String): Unit = {
		this.parts += str
	}
	def show = this.parts.foreach(println)
}
